plugins/org.eclipse.osee.framework.skynet.core/src/org/eclipse/osee/framework/skynet/core/User.java

```diff
@@ -127,18 +133,23 @@ public void toggleFavoriteBranch(BranchId favoriteBranch) {
       saveSettings();
    }
 
+   /**
+    * Cached for efficiency as this is called thousands of times to sort favorites. Cache is updated when User artifact
+    * is changed (eg: transaction in User art is different)
+    */
    public boolean isFavoriteBranch(BranchId branch) {
-      Collection<String> attributes = getAttributesToStringList(CoreAttributeTypes.FavoriteBranch);
-      for (String value : attributes) {
-         try {
-            if (branch.equals(BranchId.valueOf(value))) {
-               return true;
-            }
-         } catch (Exception ex) {
-            // do nothing
+      if (!getTransaction().equals(userModTx)) {
+         if (favoriteBranchIds == null) {
+            favoriteBranchIds = new HashSet<>(20);
+         } else {
+            favoriteBranchIds.clear();
+         }
+         for (Attribute<Object> attri : getAttributes(CoreAttributeTypes.FavoriteBranch)) {
```

Comment:
is it worth taking the time to make the FavoriteBranch attribute an AttributeTypeLong and not an AttributeTypeString to avoid converting every time we store/fetch?

Reply:
Yes, love the idea.  Want to get this in to resolve the immediate lockups and missing buttons.  Will make an action to convert which will take a little time and some validation of data in dbs.

Reply:
Sounds great!